logic/transaction_pool.js

```diff
@@ -1020,4 +1050,46 @@ __private.expireTransactions = function(transactions, cb) {
 	);
 };
 
+/**
+ * Undo unconfirmed transaction from mem account and removes transactions from the pool if it is expired.
+ *
+ * @private
+ * @param {Object[]} transactions - Array of unconfirmed transactions to be undo and expire
+ * @param {function} cb - Callback function
+ * @returns {SetImmediate} error, ids[]
+ */
+__private.expireAndUndoUnconfirmedTransactions = (transactions, cb) => {
+	async.eachSeries(
+		transactions,
+		(transaction, eachSeriesCb) => {
+			if (!transaction) {
+				return setImmediate(eachSeriesCb);
+			}
+
+			if (__private.isExpired(transaction)) {
+				modules.transactions.undoUnconfirmed(transaction, undoUnconfirmErr => {
+					if (undoUnconfirmErr) {
+						library.logger.error(
+							`Failed to undo unconfirmed transaction: ${transaction.id}`,
+							undoUnconfirmErr
+						);
+						return setImmediate(eachSeriesCb);
+					}
+					// Remove transaction from unconfirmed, queued and multisignature lists
+					self.removeUnconfirmedTransaction(transaction.id);
+					library.logger.info(
+						`Expired transaction: ${
+							transaction.id
+						} received at: ${transaction.receivedAt.toUTCString()}`
+					);
+					return setImmediate(eachSeriesCb);
+				});
+			} else {
```

Comment:
Remove this `else` part. its a good practice to not use `else` with single `if` block.